camel/models/model_factory.py

```diff
@@ -46,16 +50,18 @@ def create(
             BaseModelBackend: The initialized backend.
         """
         model_class: Any
-        if model_type.is_openai:
+        if model_platform.is_openai and model_type.is_openai:
             model_class = OpenAIModel
         elif model_type == ModelType.STUB:
             model_class = StubModel
-        elif model_type.is_open_source:
+        elif model_platform.is_opensource and model_type.is_open_source:
```

Comment:
looks like a code smell to me that there are two flags always with the same name

Reply:
update the nameing